magento/pwa-studio
# PWA-1689: Filtering Product by the attribute value

- Updated styling for filters



## Description

In scope: Text, Content, Boolean, Multi-select and Date

As a Merchant I want to show product attributes with values on PWA Storefront using native capabilities so that I manage product information without ongoing developer help

Note: The main changes are made on the boolean filter as they are now radio buttons which should behave as intended compared to their checkbox counterpart.


## Related Issue







Closes #PWA-1689

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






Preconditions:
Web store with 2 store views created
Product attributes with input types Text, Content, Boolean, Multi-select and Date created in Admin>Store>Attributes>Product
Product attributes with Boolean and Multi-select inputs are enabled for layered navigation in Admin>Store>Attributes>Used in Layered Navigation
Product attributes  with Text, Content and Date inputs aren't available for using in layered navigation 
Products that use product attributes above are created in Admin Panel
User is browsing product listing page with products mentioned above on Venia PWA Storefront

Acceptance Criteria:
User navigates to the product listing page where the products belong to and sees product attributes enabled for the layered navigation positioned there based on rank set in Store>Product>Attribute>Storefront Properties>Position. If the rank is not set to define position based on it, product attributes sorted alphabetically.
User doesn't see attributes with input type Text, Content and Date as filters in the layered navigation 
User selects product attribute value from the options in the layered navigation and see product listing updated with the product that match the selected value
Multi-select attribute values are displayed in Layered navigation filters as designed 
Boolean attribute values are displayed in layered navigation as designed 

Test Plan:
Update backend with all needed pre-conditions and verify -
Text, Content and Date are not shown in layered nav (filter group)
Verify Multi select and boolean are displayed in layered nav.
Verify Applying multi select and boolean filters gets proper results.
Verify display order is based on by position by default. And if position is not set then it should be alphabetical order
Verify above on Store view 2
Verify localized labels works
Verify existing dropdown attribute still works
Verify above. on both Category. and Search pages.
Dont see any new UI so browser testing not needed. But if any new UI gets implemented like "Pre-order" label on product then do quick check on Firefox and Safari and a mobile device. Also check how date displays in Safari and firefox.

Cypress Test coverage:
We already have filter test . But for this ticket make sure graphql is automated fully in webapi functional tests.


#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
